src/providers/dataProvider/resource-callbacks/ReferenceItemLifeCycle.ts

```diff
@@ -24,13 +25,43 @@ export default (
     return record
   },
   beforeUpdate: async (record: UpdateParams<RCOResource>) => {
+    let securityRelated: boolean | undefined
+    if (checkSecurityRelated === true) {
```

Comment:
instead of passing in a boolean for whether to test, how about passing in an optional function that does the test. Then this module does not have to know the detail of how we test different types of resource?